datafusion/src/scalar.rs

```diff
@@ -156,6 +157,58 @@ impl PartialEq for ScalarValue {
     }
 }
 
+// manual implementation of `PartialOrd` that uses OrderedFloat to
+// get defined behavior for floating point
+impl PartialOrd for ScalarValue {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        use ScalarValue::*;
+        // This purposely doesn't have a catch-all "(_, _)" so that
+        // any newly added enum variant will require editing this list
+        // or else face a compile error
```

Comment:
this comment doesn't seem to hold true anymore in this function, would be better to keep the same structure in `PartialEq` so we can also avoid catch all match here.

Reply:
Good suggestion! Updated.